compose/cli/scan_suggest.py

```diff
@@ -0,0 +1,80 @@
+import json
+import logging
+import os
+from distutils.util import strtobool
+
+from docker.constants import IS_WINDOWS_PLATFORM
+from docker.utils.config import find_config_file
+
+
+SCAN_BINARY_NAME = "docker-scan" + (".exe" if IS_WINDOWS_PLATFORM else "")
+
+log = logging.getLogger(__name__)
+
+
+class ScanConfig:
+    def __init__(self, dict):
+        self.optin = False
+        vars(self).update(dict)
+
+
+def display_scan_suggest_msg():
+    if environment_scan_avoid_suggest() or \
+            scan_already_invoked() or \
+            scan_available() is None:
+        return
+    log.info("Use 'docker scan' to run Snyk tests against images to find vulnerabilities "
+             "and learn how to fix them")
+
+
+def environment_scan_avoid_suggest():
+    return os.getenv('DOCKER_SCAN_SUGGEST', 'true').lower() == 'false'
+
+
+def scan_already_invoked():
+    docker_folder = docker_config_folder()
+    if docker_folder is None:
+        return False
+
+    scan_config_file = os.path.join(docker_folder, 'scan', "config.json")
+    if not os.path.exists(scan_config_file):
+        return False
+
+    data = ''
+    with open(scan_config_file) as f:
+        data = f.read()
+    scan_config = json.loads(data, object_hook=ScanConfig)
+    return scan_config.optin if isinstance(scan_config.optin, bool) else strtobool(scan_config.optin)
```

Comment:
what happens here if anything goes wrong (can't read file, can't unmarshall json, ...)
We should return true by default, to not bother users if we cannot check if scan has been executed previously